### ISSUE TITLE
Uncaught Error: Call to a member function addItem() on array in app/code/Magento/Sales/Model/Order/Shipment.php

### ISSUE THREAD

**Issue by raisaev** [magento/magento2#8022]
When adding a Track to a Shipment programmatically.

### Preconditions
1. Magento 2.1.1
2. PHP 5.6.24

### Steps to reproduce
1. Load or create a Shipment model

```
/** @var $shipment \Magento\Sales\Model\Order\Shipment */
$shipmentCollection = $this->_objectManager->get(
    'Magento\Sales\Model\ResourceModel\Order\Shipment\CollectionFactory'
)->create();
$shipment = $shipmentCollection->getFirstItem();
```
		
2. Make a call
`$shipment->getTracks();`
		
3. Create a Track model and try to add it to the Shipment

```
/** @var $track \Magento\Sales\Model\Order\Shipment\Track */
$track = $this->_objectManager->get(
    'Magento\Sales\Model\Order\Shipment\TrackFactory'
)->create();
$track->setNumber('Test Number')
          ->setTitle('Test Title')
          ->setCarrierCode('Test CODE');

$shipment->addTrack($track)->save();
```
### Expected result
1. Successfull save

### Actual result

You will receive an error **Call to a member function addItem() on array** because the method **getTracksCollection()** in Shipment model will return an array instead of Collection Object.

**Comment by xmav** [comment]
@raisaev Thanks for reporting this issue. 
We've created internal ticket MAGETWO-63653 to address this issue.

**Comment by titechnologies** [comment]
Is it solved yet ?

**Comment by magento-engcom-team** [comment]
@raisaev, thank you for your report.
We've created internal ticket(s) MAGETWO-63653 to track progress on the issue.

**Comment by okorshenko** [comment]
The issue has been fixed and delivered to 2.2-develop branch. Will be available with 2.2.2 release

**Comment by ferraramax** [comment]
in wich way we can solve in 2.2.1?
